File: frontend/src/app/core/services/booking.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';

export interface Booking {
  id?: number;
  checkInDate: string;
  checkOutDate: string;
  status: string;
  room: {
    id: number;
  };
  user: {
    id: number;
  };
}

@Injectable({
  providedIn: 'root',
})
export class BookingService {
  private baseUrl = `${environment.apiUrlBooking}/booking`; // ✅ utilise environment

  constructor(private http: HttpClient) {}

  createBooking(
    userId: number,
    roomId: number,
    checkInDate: string,
    checkOutDate: string
  ): Observable<string> {
    const url = `${
      this.baseUrl
    }/create/${userId}/${roomId}/${encodeURIComponent(
      checkInDate
    )}/${encodeURIComponent(checkOutDate)}`;
    return this.http.post(url, null, { responseType: 'text' });
  }

  cancelBooking(bookingId: number): Observable<string> {
    return this.http.put(`${this.baseUrl}/cancel/${bookingId}`, null, {
      responseType: 'text',
    });
  }

  getBookingsByUser(userId: number): Observable<Booking[]> {
    return this.http.get<Booking[]>(`${this.baseUrl}/user/${userId}`);
  }

  getAllBookings(): Observable<Booking[]> {
    return this.http.get<Booking[]>(`${this.baseUrl}/all`);
  }
}

File: backend/src/main/java/com/hotelManagement/services/BookingService.java
package com.hotelManagement.services;


import com.hotelManagement.pojo.Booking;
import com.hotelManagement.pojo.Room;

import java.util.List;
import java.util.Optional;

public interface BookingService {
    Booking createBooking(Integer userId, Integer roomId, String checkInDate, String checkOutDate);

    Optional<Booking> getConfirmedBookingByRoom(Room room);

    List<Booking> getAllBookings();

    List<Booking> getBookingsByUserId(Integer userId);

    void cancelBooking(Integer bookingId);

}
